tests/test_scenarios.py

```diff
@@ -0,0 +1,137 @@
+    "dereference_test3_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+    "dereference_test4_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+    "dereference_test5_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
+        {"expected_len_triples": "TBD", "mode": "nostop"},
+    ],
+}
+
+
+# launch a server with subprocess from local_server.py
+@pytest.mark.usefixtures("httpd_server_base", "store_info_sets")
+def test_scenarios(
+    httpd_server_base: str,
+    store_info_sets,
+):
+    assert httpd_server_base
+
+    for config in CONFIGS.glob("*.yml"):
+        if config.name not in SCENARIOS_OUTCOMES:
+            log.debug(f"Skipping scenario {config}")
+            continue
+        for test_case in SCENARIOS_OUTCOMES[config.name]:
+            mode = test_case["mode"]
+            log.debug(f"{mode=}")
+
+            # TODO: when mode is added in 0.0.5 add tests here
+            # for now continue
+
+            if mode == "nostop":
+                log.debug(f"Skipping scenario {config} with mode {mode}")
+                continue
+
+            for store_info in store_info_sets:
+                log.debug(f"Running scenario {config} with store {store_info}")
+                travharv = TravHarv(
+                    config,
+                    store_info,
+                )
+                travharv.process()
+                # assert travharv.error_occurred
+                # other assertions here based on the scenarios
+
+                # get the context name for the config name
+                context_name_graph = travharv.target_store._nmapper.key_to_ng(
+                    config.name
+                )
+                log.debug(f"{context_name_graph=}")
+
+                # make sparql for the test
+                sparql = QUERY_BUILDER.build_syntax(
+                    "execution_report_data.sparql",
+                    context_name_graph=context_name_graph,
+                )
+
+                log.debug(f"{store_info=}")
+
+                if store_info == ():
+                    sparql = QUERY_BUILDER.build_syntax(
+                        "execution_report_data.sparql",
+                    )
+                    execution_report_data_result = (
+                        travharv.target_store.select(sparql)
+                    )
+                    sparql_all = "SELECT ?s ?p ?o WHERE { ?s ?p ?o }"
+                else:
+                    execution_report_data_result = (
+                        travharv.target_store.select(sparql)
+                    )
+                    sparql_all = (
+                        f"SELECT ?s ?p ?o FROM <{context_name_graph}> "
+                        f"WHERE {{ ?s ?p ?o }}"
+                    )
+
+                all_triples = travharv.target_store.select(sparql_all)
```

Comment:
needlessly cumbersome, ignores that `all_triples()` method exists

Reply:
For the remote store I have other triples in there other then the ones in scenario so the all triples method of the rdfstore will not be sufficient for the remote one. Thats why I still use line 118

Reply:
ok, so the diff you seem to make is that you want all_triples from one selected named_graph only

then still this current solution is far from elegant -- would prefer to provide all_triples_for_ng(ng) in stead -- please not for 005